colossalai/checkpoint_io/utils.py

```diff
@@ -303,16 +319,19 @@ def save_state_dict(state_dict: dict, checkpoint_file_path: str, use_safetensors
         checkpoint_file_path (str): path to the checkpoint file.
         use_safetensors (bool): whether to use safetensors to save the checkpoint.
     """
+    # Move all tensors in the state_dict to CPU before saving to avoid serialization issues
+    state_dict_cpu = move_to_cpu(state_dict)
```

Comment:
Can you try `from torch.utils._pytree import tree_map()`?

Reply:
Ok, I have updated the code to use tree_map from torch.utils._pytree for transferring the state dictionary to CPU